src/saltext/vmware/states/esxi.py

```diff
@@ -145,6 +147,294 @@ def role_absent(
+            ] = "vmkernel adapter {} will be created on {} host(s) and updated on {} host(s).".format(
+                name, len(add_on_hosts), len(update_on_hosts)
+            )
+            ret["result"] = None
+        elif not add_on_hosts and not update_on_hosts:
+            ret[
+                "comment"
+            ] = "vmkernel adapter {} not created/updated on any host. No changes made.".format(name)
+            ret["result"] = True
+        else:
+            hosts_in_error = []
+            sample_exception = None
+            for action, hosts in [("add", add_on_hosts), ("update", update_on_hosts)]:
+                for host in hosts:
+                    try:
+                        func = None
+                        if action == "add":
+                            func = functools.partial(
+                                __salt__["vmware_esxi.create_vmkernel_adapter"]
+                            )
+                        else:
+                            func = functools.partial(
+                                __salt__["vmware_esxi.update_vmkernel_adapter"], adapter_name=name
+                            )
+                        ret_save = func(
+                            port_group_name=port_group_name,
+                            dvswitch_name=dvswitch_name,
+                            vswitch_name=vswitch_name,
+                            enable_fault_tolerance=enable_fault_tolerance,
+                            enable_management_traffic=enable_management_traffic,
+                            enable_provisioning=enable_provisioning,
+                            enable_replication=enable_replication,
+                            enable_replication_nfc=enable_replication_nfc,
+                            enable_vmotion=enable_vmotion,
+                            enable_vsan=enable_vsan,
+                            mtu=mtu,
+                            network_default_gateway=network_default_gateway,
+                            network_ip_address=network_ip_address,
+                            network_subnet_mask=network_subnet_mask,
+                            network_tcp_ip_stack=network_tcp_ip_stack,
+                            network_type=network_type,
+                            datacenter_name=datacenter_name,
+                            cluster_name=cluster_name,
+                            host_name=host,
+                            service_instance=service_instance,
+                        )
+                        ret["changes"].update(ret_save)
+                    except salt.exceptions.SaltException as exc:
+                        hosts_in_error.append(host)
+                        sample_exception = str(exc)
+            ret["comment"] = "vmkernel adapter {} created on {}, updated on {}.".format(
+                name,
+                ",".join(set(add_on_hosts) - set(hosts_in_error)) or "-",
+                ",".join(set(update_on_hosts) - set(hosts_in_error)) or "-",
```

Comment:
**suggestion (non-blocking)** it may not be important, but `sorted(set - set)` would produce a list that's always ordered.

Reply:
will change